### PR TITLE
ci: automate creation of GitHub releases and tags

### PR DESCRIPTION
**Pull request by galargh**
Follow up to https://github.com/filecoin-project/filecoin-ffi/pull/473#issuecomment-2246763797

### CODE REVIEW

**Review by BigLep — COMMENTED** [review]
I added some draft text for updating the release process in the README.

That said, I'm wondering if we can have the asset creation/attaching happen before the PR gets merged that way someone can see the assets before merging the PR.

**Review by rvagg — APPROVED** [review]

**Comment by galargh** [comment]
@BigLep In other repos, we can; here, the current instructions don't mention anything about asset creation or upload.

**Comment by BigLep** [comment]
> @BigLep In other repos, we can; here, the current instructions don't mention anything about asset creation or upload.

Yeah, I know the instructions don't mention anything about asset creation (doh - bummer).  But from looking at https://github.com/filecoin-project/filecoin-ffi/blob/master/.github/workflows/ci.yml#L152 , doesn't it happen on a tag?  I'm wondering if we can adjust the triggering conditions and get it into the instructions.

**Comment by BigLep** [comment]
2024-08-01 conversation with IPDX/FilOz: @galargh will assess week of 2024-08-05 to see if asset creation can be executed in the proposed flow or whether needs separate dedicated work task.

File: README.md
Comment (BigLep) [inline README.md:140]:
```suggestion
2. When the PR is opened:
    - a draft GitHub release is created with GitHub-generated release notes
    - assets are built and attached to the draft release
3. When the PR is merged
   - the corresponding version.json tag is created 
   - the GitHub release is published
```

**Review by BigLep — APPROVED** [review]
Things look good to me once we have the README correct about the new reality.

**Review by rvagg — APPROVED** [review]

**Comment by galargh** [comment]
The release process and possible improvements for the future were now described in https://github.com/filecoin-project/filecoin-ffi/pull/475/commits/a9cacd952415c1bf075401804a546a3c8acee9b1.

**Review by BigLep — COMMENTED** [review]
I have the same RELEASE.md questions/comments as had in https://github.com/filecoin-project/builtin-actors/pull/1571#pullrequestreview-2236300196 .  Otherwise it looks good to me.

File: RELEASE.md
Comment (BigLep) [inline RELEASE.md:7]:
```suggestion
1. Create a pull request which updates the `version` in the [top-level `version.json` file](https://github.com/filecoin-project/filecoin-ffi/blob/master/version.json).
   - Title the PR `build: X.Y.Z release`
```

File: RELEASE.md
Comment (BigLep) [inline RELEASE.md:12]:
```suggestion
    3. Create a draft GitHub release with the version as the tag.  (A git tag with this version string will be created when the release is published.)
```

File: RELEASE.md
Comment (BigLep) [inline RELEASE.md:20]:
```suggestion
    4. If the draft release exists, publish it. Otherwise, create and publish a new release with the version as the git tag.  Publishing the release creates the git tag.
```

**Review by BigLep — APPROVED** [review]
I added a couple of suggestions that had from https://github.com/filecoin-project/builtin-actors/pull/1571

I'm going to commit those and then merge as had verbal during 2024-08-14 Lotus maintainers call where @Stebalien said these changes are good given we don't have to do crate publishing.

**Comment by galargh** [comment]
Thank you for all the reviews, and help getting it merged 🙇

**Pull request merged**
